clients/src/test/java/org/apache/kafka/common/requests/RequestResponseTest.java

```diff
@@ -1444,43 +1450,63 @@ private DeleteGroupsResponse createDeleteGroupsResponse() {
                             .setPartitionIndex(0)
-                            .setTimestamp(1000000L)
+                            .setTimestamp(timestamp)
                             .setMaxNumOffsets(10)
                             .setCurrentLeaderEpoch(5)));
             return ListOffsetsRequest.Builder
-                    .forConsumer(false, IsolationLevel.READ_UNCOMMITTED)
+                    .forConsumer(false, IsolationLevel.READ_UNCOMMITTED, false)
                     .setTargetTimes(Collections.singletonList(topic))
                     .build((short) version);
         } else if (version == 1) {
             ListOffsetsTopic topic = new ListOffsetsTopic()
                     .setName("test")
                     .setPartitions(Arrays.asList(new ListOffsetsPartition()
                             .setPartitionIndex(0)
-                            .setTimestamp(1000000L)
+                            .setTimestamp(timestamp)
                             .setCurrentLeaderEpoch(5)));
             return ListOffsetsRequest.Builder
-                    .forConsumer(true, IsolationLevel.READ_UNCOMMITTED)
+                    .forConsumer(true, IsolationLevel.READ_UNCOMMITTED, false)
                     .setTargetTimes(Collections.singletonList(topic))
                     .build((short) version);
-        } else if (version >= 2 && version <= LIST_OFFSETS.latestVersion()) {
+        } else if (version >= 2 && version <= 6) {
             ListOffsetsPartition partition = new ListOffsetsPartition()
                     .setPartitionIndex(0)
-                    .setTimestamp(1000000L)
+                    .setTimestamp(timestamp)
                     .setCurrentLeaderEpoch(5);
 
             ListOffsetsTopic topic = new ListOffsetsTopic()
                     .setName("test")
                     .setPartitions(Arrays.asList(partition));
             return ListOffsetsRequest.Builder
-                    .forConsumer(true, IsolationLevel.READ_COMMITTED)
+                    .forConsumer(true, IsolationLevel.READ_COMMITTED, false)
                     .setTargetTimes(Collections.singletonList(topic))
                     .build((short) version);
+        } else if (version >= 7 && version <= LIST_OFFSETS.latestVersion()) {
+            ListOffsetsPartition partition = new ListOffsetsPartition()
+                .setPartitionIndex(0)
+                .setTimestamp(timestamp)
+                .setCurrentLeaderEpoch(5);
+
+            ListOffsetsTopic topic = new ListOffsetsTopic()
+                .setName("test")
+                .setPartitions(Arrays.asList(partition));
+            if (timestamp == ListOffsetsRequest.MAX_TIMESTAMP) {
+                return ListOffsetsRequest.Builder
+                        .forConsumer(true, IsolationLevel.READ_COMMITTED, false)
+                        .setTargetTimes(Collections.singletonList(topic))
+                        .build((short) version);
+            } else {
```

Comment:
We could also remove this. Note that `.forConsumer(true, IsolationLevel.READ_COMMITTED, false)` is used here. The last argument should have been `true`. It shows that it does not really bring any value.

Reply:
removed it.